xarray/tests/test_backends.py

```diff
@@ -1916,6 +1916,46 @@ def test_dump_encodings_h5py(self):
             assert actual.x.encoding['compression_opts'] is None
 
 
+@requires_h5netcdf
+@requires_dask
+@pytest.mark.filterwarnings('ignore:deallocating CachingFileManager')
+class TestH5NetCDFViaDaskData(TestH5NetCDFData):
+
+    @contextlib.contextmanager
+    def roundtrip(self, data, save_kwargs={}, open_kwargs={},
+                  allow_cleanup_failure=False):
+        with TestH5NetCDFData.roundtrip(
+                self, data, save_kwargs, open_kwargs,
+                allow_cleanup_failure) as ds:
+            yield ds.chunk()
+
+    def test_dataset_caching(self):
+        # caching behavior differs for dask
+        pass
+
+    @pytest.mark.xfail(reason="Failing to round trip unlimited dims")
+    def test_encoding_unlimited_dims(self):
+        # TODO: this should pass
+        super(TestH5NetCDFViaDaskData, self).test_encoding_unlimited_dims()
```

Comment:
I'm not sure why this isn't working. As far as I can tell, it passes when not using dask.

Reply:
I think the problem is that `ds.chunk()` does not preserve `encoding`. I don't think we test this behavior with netCDF4, either.

We should probably preserve `encoding` in chunks(), but also in the `roundtrip()` method we should trigger dask by passing `chunks={}` instead of calling `.chunk()` explicitly.